sdk/lib/rtl/rangelist.c

```diff
@@ -13,546 +13,1557 @@
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("RtlpAddIntersectingRanges: Status %X\n", Status);
+            return Status;
+        }
+    }
+
+    ASSERT((RtlEntry->PrivateFlags & RTLP_ENTRY_IS_MERGED));
+
+    CurrentRtlEntry = RtlpEntryFromLink(RtlEntry->ListEntry.Flink);
+    NextRtlEntry = RtlpEntryFromLink(CurrentRtlEntry->ListEntry.Flink);
+
+    while (&CurrentRtlEntry->ListEntry != ListHead)
+    {
+        if (AddRtlEntry->End < CurrentRtlEntry->Start)
+        {
+            break;
+        }
+
+        if (CurrentRtlEntry->PrivateFlags & RTLP_ENTRY_IS_MERGED)
+        {
+            DPRINT("RtlpAddIntersectingRanges: PrivateFlags & RTLP_ENTRY_IS_MERGED\n");
+            DbgBreakPoint();
+
+            MergedRtlEntry = RtlpEntryFromLink(CurrentRtlEntry->Merged.ListHead.Flink);
+            NextMergedRtlEntry = RtlpEntryFromLink(MergedRtlEntry->ListEntry.Flink);
+
+            while (&MergedRtlEntry->ListEntry != &CurrentRtlEntry->Merged.ListHead)
+            {
+                RemoveEntryList(&MergedRtlEntry->ListEntry);
+
+                DPRINT("RtlpAddIntersectingRanges: MergedRtlEntry %p [%I64X-%I64X]\n", MergedRtlEntry, MergedRtlEntry->Start, MergedRtlEntry->End);
+
+                Status = RtlpAddToMergedRange(RtlEntry, MergedRtlEntry, Flags);
+                ASSERT(NT_SUCCESS(Status));
+
+                MergedRtlEntry = NextMergedRtlEntry;
+                NextMergedRtlEntry = RtlpEntryFromLink(MergedRtlEntry->ListEntry.Flink);
+            }
+
+            ASSERT(IsListEmpty(&CurrentRtlEntry->Merged.ListHead));
+
+            RemoveEntryList(&CurrentRtlEntry->ListEntry);
+
+            //ExFreeToPagedLookasideList(&RtlpRangeListEntryLookasideList, CurrentRtlEntry);
+            RtlpFreeMemory(CurrentRtlEntry, 'elRR');
+        }
+        else
+        {
+            RemoveEntryList(&CurrentRtlEntry->ListEntry);
+
+            DPRINT("RtlpAddIntersectingRanges: CurrentRtlEntry %p [%I64X-%I64X]\n", CurrentRtlEntry, CurrentRtlEntry->Start, CurrentRtlEntry->End);
+
+            Status = RtlpAddToMergedRange(RtlEntry, CurrentRtlEntry, Flags);
+            ASSERT(NT_SUCCESS(Status));
+        }
+
+        CurrentRtlEntry = NextRtlEntry;
+        NextRtlEntry = RtlpEntryFromLink(CurrentRtlEntry->ListEntry.Flink);
+    }
+
+    DPRINT("RtlpAddIntersectingRanges: AddRtlEntry %p [%I64X-%I64X]\n", AddRtlEntry, AddRtlEntry->Start, AddRtlEntry->End);
+
+    Status = RtlpAddToMergedRange(RtlEntry, AddRtlEntry, Flags);
+    ASSERT(NT_SUCCESS(Status));
+
+    return Status;
+}
+
+CODE_SEG("PAGE")
+static
+NTSTATUS
+RtlpAddRange(
+    _In_ PLIST_ENTRY ListHead,
+    _In_ PRTLP_RANGE_LIST_ENTRY AddRtlEntry,
+    _In_ ULONG Flags)
+{
+    PRTLP_RANGE_LIST_ENTRY RtlEntry;
+  
+    PAGED_CODE_RTL();
+    ASSERT(AddRtlEntry);
+
+    DPRINT("RtlpAddRange: [%X] %p, %p [%I64X-%I64X]\n", Flags, ListHead, AddRtlEntry, AddRtlEntry->Start, AddRtlEntry->End);
+
+    AddRtlEntry->PublicFlags &= ~RTL_RANGE_CONFLICT;
```

Comment:
Now RtlpAddRange is called from RtlpDeleteFromMergedRange() and from RtlAddRange(). In both cases, the RTL_RANGE_CONFLICT flag is cleared.
But just in case, this is cleared directly in RtlpAddRange. A bit of insurance.